webapp/src/main/java/org/apache/atlas/web/rest/MeshEntityAssetLinkREST.java

```diff
@@ -0,0 +1,117 @@
+package org.apache.atlas.web.rest;
+
+import org.apache.atlas.RequestContext;
+import org.apache.atlas.annotation.Timed;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.instance.LinkMeshEntityRequest;
+import org.apache.atlas.repository.store.graph.AtlasEntityStore;
+import org.apache.atlas.utils.AtlasPerfMetrics;
+import org.apache.atlas.utils.AtlasPerfTracer;
+import org.apache.atlas.web.util.Servlets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
+@Path("mesh-asset-link")
+@Singleton
+@Service
+@Consumes({Servlets.JSON_MEDIA_TYPE, MediaType.APPLICATION_JSON})
+@Produces({Servlets.JSON_MEDIA_TYPE, MediaType.APPLICATION_JSON})
+public class MeshEntityAssetLinkREST {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MeshEntityAssetLinkREST.class);
+    private static final Logger PERF_LOG = AtlasPerfTracer.getPerfLogger("rest.ProductAssetLinkREST");
+
+    private final AtlasEntityStore entitiesStore;
+
+    @Inject
+    public MeshEntityAssetLinkREST(AtlasEntityStore entitiesStore) {
+        this.entitiesStore = entitiesStore;
+    }
+
+    /**
+     * Links a product to entities.
+     *
+     * @param domainGuid the ID of the product to be linked
+     * @param request    the request containing the GUIDs of the assets to link the product to
+     * @throws AtlasBaseException if there is an error during the linking process
+     */
+
+    @POST
+    @Path("/{domainId}/link-domain-to-asset")
```

Comment:
Suggestion
link-domain-to-asset -> link-domain
unlink-domain-from-asset -> unlink-domain